api/catalog/api/views/media_views.py

```diff
@@ -46,6 +46,28 @@ def __init__(self, *args, **kwargs):
             msg = "Viewset fields are not completely populated."
             raise ValueError(msg)
 
+    def check_throttles(self, request):
+        """
+        Set the rate-limit headers and copy ``super`` implementation for raising
+        throttling exceptions.
+        """
+
+        throttle_durations = []
+        for index, throttle in enumerate(self.get_throttles()):
+            if not throttle.allow_request(request, self):
```

Comment:
Rather than copying this method so that we can add the line to modify headers, do you think we could augment `allow_request` on the throttle class to do it? It'll have access to headers via the `self` parameter and could add them before returning the regular result of `allow_request`.

We'd have to change the method for finding the suffix for the headers though, as we wouldn't have the index in there. I thought we could use the `scope` name, but the TenPerDay style throttles don't have a scope. We could either add unique scopes to them or find the next suffix from the existing headers. I think I prefer the scope approach personally as it is human-readable and would probably help debugging a question from users "why am I being throttled" this way, but I don't have a strong preference.

This is also not a blocking request for a change, but I do think it would be nice to not have to keep this method copied and have it potentially drift from the upstream implementation (however unlikely that might be) or update the comment below referencing a DRF issue number to clarify that.